### DISCUSSION TITLE
Binding static members - a dumb newbie question

### DISCUSSION THREAD

**Discussion by uwu-crj** [wjakob/nanobind#790] [Q&A]
I notice the ...`static` counterparts for the various `m.def`...`()` methods. But why do they need to exist?

If, for example, nanobind added an overload to `def_rw()` for static members instead of defining `def_rw_static()`:
```c++
template<typename C, typename D, typename ...Extra>
class_ &def_rw(const char *name, D C::* p, const Extra&... extra);
template<typename D, typename ...Extra>
class_ &def_rw(const char *name, D *p, const Extra&... extra);
```
...it would Just Work and the binding would not need to be updated if whether or not a member was static later changed?

**Comment by wjakob** [comment]
Are you sure? It might work for an instance of your type, but I would guess that you then can't access it directly on the class itself. (Or if you set something there, it doesn't correctly propagate to the C++-managed value).
